src/pages/map.tsx

```diff
@@ -15,11 +16,31 @@ const MapView = () => {
   };
 
   const [activeTag, setActiveTag] = useState<Tag>();
+  const [sessions, setSessions] = useState<Session[]>([]);
+
+  const updateActiveTag = (tag: Tag) => {
+    setActiveTag(tag === activeTag ? undefined : tag);
+  };
+
+  const filteredSessions = () => {
+    if (!activeTag) return sessions;
+
+    return sessions.filter((s) => s.sessionTag === activeTag?.id);
+  };
+
+  const fetchSessions = async () => {
+    const res = await listSessions(20, true);
+    setSessions(res);
+  };
+
+  useEffectOnce(() => {
+    fetchSessions();
+  });
 
   return (
     <>
       <div className="absolute inset-0 h-full w-full bg-gradient-to-t from-gray-800 to-black">
-        <MapContainer />
+        <MapContainer sessions={filteredSessions()} />
```

Comment:
use a `useEffect` here for the filtered sessions or a `useCallback` otherwise the function gets called a lot even if the filter didn't change.